molo/core/wagtail_hooks.py

```diff
@@ -80,10 +80,24 @@ def show_reactionquestions_response_for_users_have_access(request, menu_items):
 def show_main_language_only(parent_page, pages, request):
     main_language = Languages.for_site(request.site).languages.filter(
         is_main_language=True).first()
-    specific_pages = [page.specific for page in pages]
+    if isinstance(pages, QuerySet):
+        specific_pages = pages
+        for page in pages:
+            if not page.specific:
+                specific_pages = specific_pages.exclude(pk=page.pk)
+    else:
+        specific_pages = [page.specific for page in pages]
     if pages and main_language and parent_page.depth > 2:
-        new_pages = [page for page in specific_pages
-                     if page.language and page.language.pk == main_language.pk]
+        if isinstance(specific_pages, QuerySet):
+            new_pages = specific_pages
+            for page in specific_pages:
+                if not ((page.language and page.language.pk) ==
+                        main_language.pk):
```

Comment:
I think your brackets are incorrect here right now. This doesn't make any sense and I'm surprised it's passing. Please make sure this code is actually being run in the tests.
This says "(if this page has a language and that language has a pk) and the main_language pk matches that True/False result then ..."
It should say "if this page has a language and (that language pk matches the main language pk) then..."

Reply:
@sewagodimo this is still wrong.
It should be `if not (page.language and page.language.pk == main_language.pk):`